core/Http.php

```diff
@@ -430,7 +439,13 @@ public static function sendHttpRequestBy(
                 fclose($handle);
             } else {
                 $response = @file_get_contents($aUrl, 0, $ctx);
-                if ($response === false) {
+
+                // try to get http status code from response headers
+                if (isset($http_response_header) && preg_match('~^HTTP/(\d\.\d)\s+(\d+)(\s*.*)?~', implode("\n", $http_response_header), $m)) {
+                    $status = (int)$m[2];
+                }
+
+                if (!$status && $response === false) {
```

Comment:
403 with no response is a valid answer from the server. Throwing an general exception makes it impossible to react proper. In the translation command the transifex API is queried. It requires HTTP-Auth. If the given username & password is invalid the API returns a 401. Before this change there was an exception thrown, but without a proper reason.
I see two proper solutions, one is that above, the other is to throw proper exceptions based on the status code. e.g. AuthenticationFailedException for 401, ForbiddenException for 403 and so on